tests/cxx/default_tpl_arg.cc

```diff
@@ -43,6 +43,18 @@ struct Outer2 {
 // IWYU: IndirectTemplate needs a declaration
 Outer2<int, IndirectTemplate<int>> o2;
 
+// IWYU: IndirectClass is...*indirect.h
+template <typename T = IndirectClass>
+// IWYU: IndirectClass is...*indirect.h
+void FnWithDefArgs(T = IndirectClass{}) {
```

Comment:
I think it would make more sense with `T = T()` (or `T = T{}` if you insist on the curlies :smile:) -- for two reasons:

* No need to `// IWYU:` capture the use
* Probably more closely mirrors what a real-world usecase would look like

Oh, by the way, the graveljp example also had the equivalent of `T = {}`, which might be nice to test separately.